heron/api/src/java/org/apache/heron/streamlet/impl/operators/CustomOperator.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.heron.streamlet.impl.operators;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.heron.api.bolt.OutputCollector;
+import org.apache.heron.api.topology.TopologyContext;
+import org.apache.heron.api.tuple.Tuple;
+import org.apache.heron.api.tuple.Values;
+
+/**
+ * CustomOperator is the base class for all user defined operators.
+ * Usage:
+ * 1. Create user defined operator
```

Comment:
This part needs further discussion. IMO, users should only need to provide the Bolt to the streamlet API instead of having direct access to streamlet's operator implementation and being required to implement an additional `process` method.

Once the desired bolt is provided, streamlet should take care of the remaining stuff.

Reply:
It is not an addition method users need to implement.

This class is for all "new" user defined streamlet operators and there is no Bolt concept in it.

Existing Bolts (or write operators in low level way) is supported but it another different use case.
```

class UserBoltOperator
    extends UserBolt
    implements ICustomOperator<InputType, OutputType> {
}

```
Then the bolt is converted to a streamlet operator and can be used in Streamlet API.
```
source
   ....
  .perform(new UserBoltOperator())
   ....

```